Assets/Scripts/Game/HullManager.cs

```diff
@@ -160,6 +161,50 @@ private void GenerateFloor()
 			});
 		}
 
+		/// <summary>
+		/// Generates the walls for each module;
+		/// </summary>
+		[ButtonGroup]
+		public void GenerateModuleWalls()
+		{
+			hullData.GenerateModuleSizes();
+			SetupMasterGroup();
+
+			if (moduleGroup != null)
+				Destroy(moduleGroup.gameObject);
+
+			moduleGroup = new GameObject(nameof(moduleGroup));
+			moduleGroup.transform.position = transform.position;
+			moduleGroup.transform.parent = hullGroup.transform;
+
+			moduleGroup.transform.rotation = Entities.PlayerShipEntity.FindObjectOfType<PlayerShipEntity>().gameObject.transform.rotation;
```

Comment:
Not a problem or anything that you need to fix, but just as an FYI, I think you can go PlayerController.Instance.PlayerShipEntity.

Reply:
Thanks, I should've commented that this needed to be re-written / find another way. It was a quick fix for aligning to the direction the ship is currently facing